devdoc/constbuffer_requirements.md

```diff
@@ -78,7 +78,13 @@ MOCKABLE_INTERFACE(constbuffer,
 
     FUNCTION(, CONSTBUFFER_TO_FIXED_SIZE_BUFFER_RESULT, CONSTBUFFER_to_fixed_size_buffer, CONSTBUFFER_HANDLE, source, unsigned char*, destination, uint32_t, destination_size, uint32_t*, serialized_size),
 
-    FUNCTION(, CONSTBUFFER_FROM_BUFFER_RESULT, CONSTBUFFER_from_buffer, const unsigned char*, source, uint32_t, size, uint32_t*, consumed, CONSTBUFFER_HANDLE*, destination)
+    FUNCTION(, CONSTBUFFER_FROM_BUFFER_RESULT, CONSTBUFFER_from_buffer, const unsigned char*, source, uint32_t, size, uint32_t*, consumed, CONSTBUFFER_HANDLE*, destination),
+
+    FUNCTION(, CONSTBUFFER_HANDLE, CONSTBUFFER_create_writable_handle, uint32_t, size),
```

Comment:
>CONSTBUFFER_HANDLE [](http://example.com/codeflow?start=15&length=18)

Looking at this I am wondering if we could have something to prohibit accidentally passing a writeable const buffer handle to other APIs (like passing a writeable CONST BUFFER to a constbuffer array for example)

Think something like a CONSTBUFFER_WRITEABLE_HANDLE and the compiler would not accept that when passed to constbuffer_array APIs for example.
I don't have a solution right now and here (typedef CONSTBUFFER_HANDLE CONSTBUFFER_WRITEABLE_HANDLE will not make the compiler cry, but would be a nice thing to have. 
Maybe something where CONSTBUFFER_HANDLE has a different qualifier of sorts ...?
Just thinking our loud

Reply:
Created two structures CONSTBUFFER_HANDLE_DATA_TAG and CONSTBUFFER_WRITABLE_HANDLE_DATA_TAG with help of macro `MU_DEFINE_STRUCT` with the same fields. Now on compile time if the user by mistake passes one structure type instead of the other then the build will fail, and no run-time check is required.